publisher/publisher.go

```diff
@@ -151,6 +157,12 @@ func WithNotify(readyCh chan struct{}, unreadyCh chan error) Option {
 	}
 }
 
+func WihConfirmation(confirmation bool) Option {
```

Comment:
typo `WihConfirmation` -> `WithConfirmation`

Reply:
```go
func WithConfirmation(buffer int) Option {
```

buffer is the conformationChs size
validate that buffer is greater than 1. 
make conformationChs with given size